apps/vs-code-designer/src/app/commands/deploy/hybridLogicApp/index.ts

```diff
@@ -88,6 +101,157 @@ export const deployHybridLogicApp = async (context: IActionContext, node: SlotTr
   }
 };
 
+export const deployHybridLogicAppV2 = async (context: IActionContext, node: SlotTreeItem, effectiveDeployFsPath: string) => {
+  try {
+    await window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        title: localize('deployingHybridLogicAppV2', 'Deploying hybrid logic app (v2)'),
+        cancellable: true,
+      },
+      async (progress) => {
+        context.telemetry.properties.lastStep = 'zipAndDeploy';
+
+        progress.report({ increment: 20, message: 'Zipping content for deployment' });
+
+        // Step 1: Create the zip file on disk
+        const zipFilePath = await createZipFileOnDisk(effectiveDeployFsPath);
+
+        // Step 2: Get the access token
+        const accessToken = await getAccessTokenForZipDeploy(node, context as ILogicAppWizardContext);
+
+        if (!node.hybridSite.configuration?.ingress?.fqdn) {
+          const clientContainer = await createContainerClient(context as ILogicAppWizardContext);
+          await waitForIngressFqdn(node, clientContainer);
+        }
+
+        progress.report({ increment: 40, message: `Waiting for logic app to be ready: ${node.hybridSite.configuration.ingress.fqdn} ` });
+
+        await waitForContainerAppProvisioning(node.hybridSite.configuration.ingress.fqdn);
+
+        progress.report({ increment: 40, message: 'Uploading zip to Logic App' });
+        // Step 3: Call the /zipDeploy API
+        const zipDeployUrl = `https://${node.hybridSite.configuration.ingress.fqdn}/api/zipDeploy`;
+        await axios.put(zipDeployUrl, fs.createReadStream(zipFilePath), {
+          headers: {
+            Authorization: `Bearer ${accessToken}`,
+            'Content-Type': 'application/zip',
+          },
+          maxContentLength: Number.POSITIVE_INFINITY,
+          maxBodyLength: Number.POSITIVE_INFINITY,
+        });
+
+        progress.report({ increment: 40, message: 'Deployment completed successfully' });
+
+        // Clean up the zip file after deployment
+        fs.unlinkSync(zipFilePath);
+      }
+    );
+  } catch (error) {
+    throw new Error(`${localize('errorDeployingHybridLogicAppV2', 'Error deploying hybrid logic app (v2)')} - ${error.message}`);
+  }
+};
+
+const waitForContainerAppProvisioning = async (fqdn: string): Promise<void> => {
+  const maxRetries = 40; // Maximum number of retries
+  const delay = 3000; // Delay between retries in milliseconds (3 seconds)
+  const zipDeployUrl = `https://${fqdn}/api/zipDeploy`;
+
+  for (let attempt = 0; attempt < maxRetries; attempt++) {
+    try {
+      const response = await axios.get(zipDeployUrl, { timeout: 3000 });
+      if (response.status === 200) {
+        return; // Container app is ready
+      }
+    } catch (_) {
+      // Ignore errors and retry
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, delay)); // Wait before retrying
+  }
+
+  throw new Error('Logic app is not ready to serve traffic after waiting.');
```

Comment:
Can you localize the error and also log the error in the 
```typescript
context.telemetry.properties.getAccessTokenError = errorMessage;
```
getAccessTokenError its just an example it can be any value